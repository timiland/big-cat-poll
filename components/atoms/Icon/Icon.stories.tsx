import IconSizeEnum from '@models/enums/IconSizeEnum';
import { ComponentStory } from '@storybook/react';
import Icon from './Icon';

export default {
  title: 'Components/Atoms/Icon',
  component: Icon,

  parameters: {
    docs: {
      description: {
        component: `More information about adding icons to the project in the Iconography documentation.`,
      },
    },
  },
};

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const Default = Template.bind({});

Default.args = {
  size: IconSizeEnum.xl,
  name: 'vimeo',
  colorClassName: '',
  className: '',
};
