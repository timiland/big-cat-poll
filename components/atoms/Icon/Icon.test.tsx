import { composeStories } from '@storybook/testing-react';
import { render } from '@utils/jest.utils';
import * as stories from './Icon.stories';

const { Default } = composeStories(stories);

describe('Icon', () => {
  it('should render', async () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
