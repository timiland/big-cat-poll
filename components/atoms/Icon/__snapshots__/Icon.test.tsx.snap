// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render 1`] = `
<svg
  class="vimeo"
  color=""
  style="display: inline-block; stroke: currentColor; fill: currentColor; width: 1.75rem; height: 1.75rem;"
  viewBox="0 0 1024 1024"
>
  <path
    d="M512 0c-282.76 0-512 229.229-512 512 0 282.769 229.24 512 512 512s512-229.231 512-512c0-282.771-229.239-512-512-512zM873.967 343.551c-3.204 70.543-52.527 167.182-147.802 289.819-98.492 128.111-181.858 192.166-250.086 192.166-42.206 0-77.939-38.997-107.156-117.043-19.495-71.518-39.011-143.006-58.496-214.532-21.653-78.002-44.909-117.055-69.8-117.055-5.405 0-24.408 11.429-56.893 34.161l-34.067-43.945c35.732-31.396 71.025-62.854 105.731-94.344 47.717-41.201 83.502-62.896 107.397-65.096 56.39-5.416 91.106 33.176 104.141 115.694 14.076 89.085 23.808 144.491 29.278 166.157 16.247 73.903 34.15 110.822 53.658 110.822 15.177 0 37.995-24 68.352-71.898 30.34-47.945 46.596-84.446 48.785-109.449 4.326-41.389-11.942-62.121-48.785-62.121-17.37 0-35.273 3.96-53.645 11.89 35.605-116.678 103.654-173.383 204.109-170.178 74.483 2.189 109.564 50.503 105.28 144.952z"
  />
</svg>
`;
